### PR TITLE
Form

### PR DESCRIPTION
**Pull request by macfrei**
I implemented a multi step form which renders a group of radio buttons one after another. 

- Form data is currently only logged out, logic to determine the house will be in a different PR
- There is only basic styling, this will also be implemented in a different PR
- Unsure how to test a `fieldset` correctly: according to my research, it is the right accessible and semantic HTML to use, however, I can't seem to get it through `getByRole` or anything like that

### CODE REVIEW

File: src/components/Form.spec.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        ]}
+        onSaveHouse={noop}
+      />
+    )
+    expect(screen.getByLabelText('Dawn')).toBeInTheDocument()
+    expect(screen.getByLabelText('Dusk')).toBeInTheDocument()
+    expect(screen.getByRole('button', { name: /submit/i })).toBeInTheDocument()
+  })
+
+  it('should render a button with text "submit" when at last question and call onSaveHouse with form data', () => {
```
Comment (jamarob) [inline src/components/Form.spec.js]: I think these are 2 tests, rendering the next question and checking for correct rendering of the last question?

File: src/components/Form.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      options: PropTypes.arrayOf(PropTypes.string),
+      id: PropTypes.number,
+    })
+  ).isRequired,
+  onSaveHouse: PropTypes.func.isRequired,
+}
+
+export default function Form({ questions, onSaveHouse }) {
+  const [formData, setFormData] = useState({})
+  const [steps, setSteps] = useState(0)
```
Comment (jamarob) [inline src/components/Form.js]: Maybe rename it to `currentStep` to prevent confusion with lastStep?

File: src/components/Form.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+  const totalSteps = questions.length - 1
+
+  return (
+    <FormStyled onSubmit={handleSubmit}>
+      {questions.map(({ question, options, id }, index) => (
```
Comment (jamarob) [inline src/components/Form.js]: Imho the `Question` component is a little bit overloaded. I would pull the logic for displaying the correct question up into this Form. Could be as simple as 
```javascript
const currentQuestion = questions[currentStep]
```

File: src/components/Form.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+      [name]: value,
+    })
+  }
+
+  function handleSubmit(event) {
+    event.preventDefault()
+    onSaveHouse(formData)
+  }
+
+  function updateSteps() {
```
Comment (jamarob) [inline src/components/Form.js]: The update is always: go to the next step, so maybe rename to `advanceStep` or something?

File: src/components/Form.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    setSteps(steps => steps + 1)
```
Comment (jamarob) [inline src/components/Form.js]: Why did you choose a functional update here?

File: src/components/Question.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+          </label>
+        ))}
+      </QuestionStyled>
+      {currentStep !== totalSteps ? (
+        <button type="button" onClick={onUpdateSteps}>
+          Next
+        </button>
+      ) : (
+        <button>Submit</button>
+      )}
```
Comment (jamarob) [inline src/components/Question.js]: Imho this belongs into the Form component. Why should the question be responsible for submitting the form?

File: src/components/Question.js [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+}
+
+export default function Question({
+  question,
+  options,
+  currentStep,
+  totalSteps,
+  steps,
+  onUpdateSteps,
+  onChange,
```
Comment (jamarob) [inline src/components/Question.js]: `currentStep`, `totalSteps` and `steps` are all realted to displaying the current question, why should the `Question` component care about other `Question` components? Imho you can make your life easier by handling the switching logic at the same level where the `questions` state lives (and then put it all into a custom hook :) )

File: src/components/Question.spec.js [hunk 8]
```diff
@@ -0,0 +1,64 @@
+import { render, screen } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import Question from './Question'
+
+describe('Question', () => {
+  it('should render a question, options and a button', () => {
+    render(
+      <Question
```
Comment (jamarob) [inline src/components/Question.spec.js:10]: Test fails with prop types warnings, you forgot to add `onChange` and `onUpdateSteps`.. Maybe think about adding CI pipeline to automatically run your tests on push?

**Review by jamarob — COMMENTED** [review]
Hi Lene 👋 Can't wait to find out which house I belong to :) I left my 2 cents in the comments, hope it is useful.

>Unsure how to test a fieldset correctly: according to my research, it is the right accessible and semantic HTML to use, however, I can't seem to get it through getByRole or anything like that

What do you want to test? You can select the fieldset with
```javascript
screen.getByRole("group", {name: /text in the legend field/i})
```

File: src/components/Form.js (hunk 3 above)
Comment (macfrei) [reply src/components/Form.js]: Thanks! I will have a look 🙂

Comment (macfrei) [reply src/components/Form.js]: Such a good recommendation! Thank you! 👍

File: src/components/Form.js (hunk 5 above)
Comment (macfrei) [reply src/components/Form.js]: I sometimes feel like I have a better overview with a functional update. But yea, maybe it's a bit too short.

File: src/components/Question.js (hunk 6 above)
Comment (macfrei) [reply src/components/Question.js]: Yep, that's correct! Thanks for pointing it out!

File: src/components/Question.js (hunk 7 above)
Comment (macfrei) [reply src/components/Question.js]: Noooice! Changed it! Will work on the custom hook next!

File: src/components/Form.spec.js (hunk 1 above)
Comment (macfrei) [reply src/components/Form.spec.js:26]: True. Don't need those tests anyway, after I refactored the `Question` component.

File: src/components/Question.spec.js (hunk 8 above)
Comment (macfrei) [reply src/components/Question.spec.js:10]: That's a really good idea! Thanks 👍

File: src/components/Form.spec.js (hunk 1 above)
Comment (macfrei) [reply src/components/Form.spec.js]: True. Will fix it.

**Pull request merged**